app/models/document.rb

```diff
@@ -2,11 +2,14 @@ class Document
   include ActiveModel::Model
   include ActiveModel::Validations
 
-  attr_accessor :content_id, :base_path, :title, :summary, :body, :format_specific_fields, :public_updated_at, :state, :bulk_published, :publication_state, :minor_update, :change_note
+  attr_accessor :content_id, :base_path, :title, :summary, :body, :format_specific_fields, :public_updated_at, :state, :bulk_published, :publication_state, :change_note
+  attr_writer :change_history, :update_type
 
   validates :title, presence: true
   validates :summary, presence: true
   validates :body, presence: true, safe_html: true
+  validates :update_type, presence: true, if: :live?
+  validates :change_note, presence: true, if: Proc.new { |d| d.update_type == 'major' && d.published? }
```

Comment:
Would be better to have this check as an explicit method, eg `requires_change_note`